tests/framework/clients/topic.go

```diff
@@ -62,3 +63,48 @@ func (t *TopicOperation) CheckTopic(topicName string) bool {
 	logger.Infof("topic ARN is %q", topicARN)
 	return true
 }
+
+func (t *TopicOperation) CreateHTTPServer() error {
+	deployment := `
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-http-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app:  sample-http-server
+  template:
+    metadata:
+      labels:
+        app: sample-http-server
+    spec:
+      containers:
+      - name: sample-http-site
+        image: quay.io/jthottan/pythonwebserver:latest
```

Comment:
Is it okay to custom made http server for this test case

Reply:
maybe you should add the server's source code + dockerfile?

Reply:
What is the image doing? Is it just a simple python HTTP server? Or is it doing more?

Reply:
@leseb : It  is a HTTP server which logs the request it receives as following
```
Body:
{"Records":[{"eventVersion":"2.2","eventSource":"ceph:s3","awsRegion":"","eventTime":"2022-02-03T09:17:44.358971Z","eventName":"ObjectCreated:Put","userIdentity":{"principalId":"ceph-user-80VCECP3"},"requestParameters":{"sourceIPAddress":""},"responseElements":{"x-amz-request-id":"8d63fb27-5af5-4af4-bad4-f9824dd66578.44397.1051013022200631677","x-amz-id-2":"ad6d-my-store-my-store"},"s3":{"s3SchemaVersion":"1.0","configurationId":"my-notification","bucket":{"name":"ceph-bkt-not","ownerIdentity":{"principalId":"ceph-user-80VCECP3"},"arn":"arn:aws:s3:::ceph-bkt-not","id":"8d63fb27-5af5-4af4-bad4-f9824dd66578.14324.3"},"object":{"key":"new","size":6,"eTag":"b1946ac92492d2347c6235b4d2611184","versionId":"","sequencer":"B89DFB61C1E9EB15","metadata":[{"key":"x-amz-content-sha256","val":"5891b5b522d5df086d0ff0b110fbd9d21bb4fc7163af34d08286a2e846f6be03"},{"key":"x-amz-date","val":"20220203T091744Z"}],"tags":[]}},"eventId":"1643879864.367782.b1946ac92492d2347c6235b4d2611184","opaqueData":"my@email.com"}]}
```
So that I can check for event type and file name from the pod logs

Reply:
I guess what would be the difference between this container and just a busybox or something else that runs a Python SimpleHTTPServer as its entrypoint.

Reply:
@leseb: are referring to why I can just python-script in a container than deploy it via deployment file using the image?
Honestly, I didn't try that way.So instead of kubectl commands, I need to run docker commands(assuming docker is used instead of podman)

Reply:
No, you can still use the Deployment object but another more common image and simply override its entrypoint with the python command. That's what I have in mind

Reply:
the issue is that the command is not a oneliner.
it is an actual script.
to make sure this is not an entirely "black-box", please add the actual code of the script together with the docker file.
in this way, if someone wants to change it and upload it somewhere else - they can.

in the future, we may move the image building process into the CI.
but once we have the script and docker file, it should be possible to do.